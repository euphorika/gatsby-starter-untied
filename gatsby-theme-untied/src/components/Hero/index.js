import React from 'react'
import PropTypes from 'prop-types'

import { Link } from 'gatsby'
import Img from 'gatsby-image'

import styles from './styles.module.styl'

const Hero = ({ children, headline, button, imgFluid }) => (
  <section className={styles.heroSection}>
    <div className={styles.background}>
      <Img
        fluid={imgFluid}
        style={{ height: '100%' }}
        imgStyle={{ objectFit: 'cover' }}
      />
    </div>
    <div className={styles.info}>
      <h2>{headline}</h2>
      <p>{children}</p>
      <div>
        <button className={styles.button}>
          <Link to={button.link}>
            <span className={styles.text}>{button.text}</span>
          </Link>
        </button>
      </div>
    </div>
  </section>
)

Hero.propTypes = {
  children: PropTypes.node.isRequired,
  headline: PropTypes.string.isRequired,
  button: PropTypes.shape({
    text: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
  }),
  imgFluid: PropTypes.object.isRequired,
}

export default Hero
