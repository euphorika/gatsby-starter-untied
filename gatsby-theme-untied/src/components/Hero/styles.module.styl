@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

.hero-section
  display: flex
  align-items: center
  position relative
  width: 100vw
  height: 100vh
  color secondary-color
  padding: 20px

  +above('m')
    padding: 45px

  .background
    position absolute
    top 0
    right 0
    bottom 0
    left 0

  .info
    position relative
    max-width: 400px
    z-index 10

    h2 {
      font-size: 18px
      margin-bottom: 15px
    }

    p {
      font-size: 12px
    }

    .button
      background-color primary-color
      width: 100%
      max-width: 300px
      border: 0
      display inline-block
      border-radius 5px
      font-size: 12px
      padding 15px

      .text
        color secondary-color
