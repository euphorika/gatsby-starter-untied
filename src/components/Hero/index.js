import React from 'react'

import styles from './styles.module.styl'

const Hero = () => (
  <section className={styles.heroSection}>Hero Section</section>
)

export default Hero
