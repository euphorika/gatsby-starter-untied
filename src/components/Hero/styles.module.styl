@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

:root
  --section-bg-color bg-color

.hero-section
  background-color var(--section-bg-color)
  height 250px
  line-height 25px
  margin 20px
  max-width 3000px
  padding 40px 100px

  .button
    background-color primary-color
    display inline-block
    borderRadius 3px
    border 0
    cursor pointer
    fontSize 18px
    padding 16px 24px
    minWidth 200px

.text
  color button-color



