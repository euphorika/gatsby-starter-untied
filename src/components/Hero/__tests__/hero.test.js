import React from 'React'
import renderer from 'react-test-renderer'
import Hero from '../'

describe('Hero', () => {
  it('renders correctly', () => {
    const imgFluid = {
      aspectRatio: 1.0,
      basde64: 'String',
      sizes: '(max-width: 4928px) 100vw, 4928px',
      src: 'image.jpg',
      srcSet: 'srcSet.image.jpg',
    }

    const button = {
      text: 'Text',
      link: '/',
    }

    const hero = (
      <Hero headline="Title" button={button} imgFluid={imgFluid}>
        Body
      </Hero>
    )
    const tree = renderer.create(hero).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
