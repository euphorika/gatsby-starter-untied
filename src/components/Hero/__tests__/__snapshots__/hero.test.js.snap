// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
<section
  className="heroSection"
>
  <div
    className="background"
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "height": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": "100%",
            "width": "100%",
          }
        }
      />
      <img
        alt=""
        src="String"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transition": "opacity 0.5s",
            "transitionDelay": "0.25s",
            "width": "100%",
          }
        }
      />
      <picture>
        <img
          alt=""
          onLoad={[Function]}
          sizes="(max-width: 4928px) 100vw, 4928px"
          src="image.jpg"
          srcSet="srcSet.image.jpg"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 0.5s",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><img sizes=\\"(max-width: 4928px) 100vw, 4928px\\" srcset=\\"srcSet.image.jpg\\" src=\\"image.jpg\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </div>
  <div
    className="info"
  >
    <h2>
      Title
    </h2>
    <p>
      Body
    </p>
    <div>
      <button
        className="button"
      >
        <Link
          to="/"
        >
          <span
            className="text"
          >
            Text
          </span>
        </Link>
      </button>
    </div>
  </div>
</section>
`;
